regparser/tree/xml_parser/paragraph_processor.py

```diff
@@ -122,6 +125,18 @@ def process(self, xml, root):
             markers = [node.label[0] for node in nodes]
             constraints = self.additional_constraints()
             depths = derive_depths(markers, constraints)
+
+            if not depths:
+                logging.warning("""Could not derive paragraph depths. Retrying
+                                   with relaxed constraints.""")
+                additional_constraints = [optional_rules.star_new_level,
+                                          optional_rules.limit_paragraph_types(
+                                           markers.lower, markers.upper,
+                                           markers.ints, markers.roman,
+                                           markers.em_ints, markers.em_roman,
+                                           markers.stars, markers.markerless)]
```

Comment:
Probably want to include this subset of reduced constraints where they were originally defined, `regparser.tree.xml_parser.reg_text:RegtextParagraphProcessor`.

Something like
```python
deemphasized = [mtypes.deemphasize(m) for m in markers]
relaxed = self.relaxed_constraints()

if not depths and deemphasized != markers:
    logging.warning("Could not derive paragraph depths. Retrying "
                   "without emphasis.")
    depths = derive_depths(deemphasize, constraints)
if not depths and relaxed:
    logging.warning("Still could not derive paragraph depths. "
                    "Retrying with relaxed constraints.")
    depths = derive_depths(deemphasized, relaxed)
```